contracts/gauges/gauge/src/msg.rs

```diff
@@ -39,8 +40,11 @@ pub struct GaugeConfig {
     pub max_available_percentage: Option<Decimal>,
     /// If set, the gauge can be reset periodically, every `reset_epoch` seconds.
     pub reset_epoch: Option<u64>,
+    /// If set, the gauge will disable itself after this many epochs. This count will not be reset if `reset_epoch` is set.
+    pub total_epochs: Option<u64>,
```

Comment:
Hmmm, this smells like some edge cases here. What happens if both `total_epochs` and `reset_epoch` are both set?

Reply:
Please make sure there is a test for this.

Reply:
test added 👍